test_data/rules/exhaustive_cases.dart

```diff
@@ -111,3 +111,39 @@ void ae(ActualEnum e) {
       print('e');
   }
 }
+
+// Some fields are deprecated.
+class DeprecatedFields {
+  final int i;
+  const DeprecatedFields._(this.i);
+
+  @deprecated
+  static const oldFoo = newFoo;
+  static const newFoo = DeprecatedFields._(1);
+  static const bar = DeprecatedFields._(2);
+  static const baz = DeprecatedFields._(3);
+}
+
+void dep(DeprecatedFields e) {
```

Comment:
What happens with the following:
```dart
switch (e) {
  case DeprecatedFields.oldFoo:
  case DeprecatedFields.bar:
  case DeprecatedFields.baz:
    break;
}
```
I suspect that we'll get a lint (because we'll be expecting `newFoo` but won't find it), but I don't think we ought to.

If we do get a lint, then I think we'll need to implement the more general solution from the outset. Rather than computing a `Map<DartObject, FieldElement>`, just compute a `Map<DartObject, List<FieldElement>>` and remove the key if any element in the list is found. If the map isn't empty by the end of the `switch` then report a lint.

Reply:
It would report a lint. I've refactored to account for this.